dali/python/nvidia/dali/pipeline.py

```diff
@@ -284,29 +284,66 @@ def outputs(self):
         self._gpu_batches_to_consume -= 1
         return self._outputs()
 
-    def _share_outputs(self):
+    def schedule_run(self):
+        """Run the pipeline without returning the resulting buffers.
+        
+        If the pipeline was created with `exec_pipelined` option set to `True`,
+        this function will also start prefetching the next iteration for
+        faster execution. It provides better control to the user about when he
+        wants to run the pipeline, when he wants to obtain resulting buffers
+        and when they can be returned to DALI pool when the results have been consumed.
+        Needs to be used together with :meth:`nvidia.dali.pipeline.Pipeline.release_outputs`
+        and :meth:`nvidia.dali.pipeline.Pipeline.share_outputs`.
+        Should not be mixed with :meth:`nvidia.dali.pipeline.Pipeline.run` in the same pipeline"""
+        if self._first_iter and self._exec_pipelined:
+            self._prefetch()
+        else:
+            self._run_once()
```

Comment:
Do we want to allow to use _run_once as well?

Reply:
I don't think so as it is rather coupled with  _run_once. I would keep it as an internal API, unless you see strong reason to expose it as well.

Reply:
Fair enough, I thought that plugins use it, but they actually use one now called `schedule_run` .